File: army/Unit/Werewolf.cpp
#include "Werewolf.h"

Werewolf::Werewolf(std::string nickName, int damage, int hitPoints, std::string title) : Unit(nickName,damage, hitPoints, title, false, false, true) {}

Werewolf::~Werewolf() {}

void Werewolf::transformInToWolf() {
    int temp = this->getHitPoints() * 1.5;

    this->state = this->wolfState;
    this->state->setCurrentHp(temp);
}

void Werewolf::transformBack() {
    int temp = this->getHitPoints() / 1.5;
    
    this->state = this->normalState;
    this->state->setCurrentHp(temp);
}

File: army/JavaArmy/src/com/maksklaban/armymax/units/Werewolf.java
package com.maksklaban.armymax.units;

import com.maksklaban.armymax.exceptions.*;

public class Werewolf extends Unit {
    public Werewolf(String nickName, int damage, int hitPoints, String title) throws CloneNotSupportedException {
        super(nickName, damage, hitPoints, title, false, false, true);
    }

    public Werewolf(String nickName) throws CloneNotSupportedException {
        this(nickName, 25, 200, "Werewolf");
    }

    public void transformInToWolf() throws ArmyException {
        int temp = this.getHitPoints() + (this.getHitPoints() / 2);

        this.state = this.wolfState;
        this.state.setCurrentHp(temp);
    }

    public void transformBack() throws ArmyException {
        int temp = this.getHitPoints() - (this.getHitPoints() / 2);
        
        this.state = this.normalState;
        this.state.setCurrentHp(temp);
    }
}
